### PR TITLE
fix docgen for subcommands

### PR DESCRIPTION
**Pull request by ajm188**
## Description

In #13015, @mattlord found two issues with the docgen tool.

1. The titles were wrong. This was an easy fix to the frontmatter.
2. The links between parent and child command pages were wrong. This was ... a whole thing, mostly due to the problem that the `linkHandler` function has no context (i.e. you know only what page you are generating a link _to_, not what you are linking to that page _from_). So, two changes here:
    a. restructure the child command files so they sit below their parent rather than having everything in a single top-level file structure
    b. run a `sed` on each child page to rewrite the incorrect link to its parent to simply go to `../`.

I ran this on top of Matt's PR and clicking around worked. I can push those changes directly to his website branch as proof, or he can pull this to double-verify; I am happy to do whatever is most preferred.

## Related Issue(s)

#13015

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/cmd/internal/docgen/docgen.go [hunk 1]
```diff
@@ -106,16 +189,19 @@ func frontmatterFilePrepender(filename string) string {
...
+	cmdName = strings.ReplaceAll(cmdName, "_", " ")
+
 	return fmt.Sprintf(frontmatter, cmdName, root)
 }
 
 func linkHandler(filename string) string {
-	name := filepath.Base(filename)
-	base := strings.TrimSuffix(name, filepath.Ext(name))
+	base := filepath.Base(filename)
+	name := strings.TrimSuffix(base, filepath.Ext(base))
```
Comment (ajm188) [inline go/cmd/internal/docgen/docgen.go:199]: these names ended up being more clear to me ("why is the thing we get from `.Base` not called base?")

File: go/cmd/internal/docgen/docgen.go [hunk 2]
```diff
@@ -79,9 +82,89 @@ func GenerateMarkdownTree(cmd *cobra.Command, dir string) error {
...
+				return fmt.Errorf("failed to move child command %s to its parent's dir: %w", fullCmdFilename, err)
+			}
+
+			sed := newParentLinkSedCommand(name, newName)
+			if out, err := sed.CombinedOutput(); err != nil {
+				return fmt.Errorf("failed to rewrite links to parent command in child %s: %w (extra: %s)", newName, err, out)
+			}
+
+			if err := os.Remove(newName + ".bak"); err != nil {
+				log.Warningf("failed to remove backup file for %s, use caution when staging files for commit: %w", newName+".bak", err)
```
Comment (ajm188) [inline go/cmd/internal/docgen/docgen.go:153]: failing to cleanup artifacts didn't feel exit-worthy to me

File: go/cmd/internal/docgen/docgen.go [hunk 3]
```diff
@@ -79,9 +82,89 @@ func GenerateMarkdownTree(cmd *cobra.Command, dir string) error {
...
+becomes
+
+_index.md
+vtctldclient_AddCellInfo.md
+vtctldclient_movetables/
+	_index.md
+	vtctldclient_movetables_show.md
+*/
+
+func restructure(rootDir string, dir string, name string, commands []*cobra.Command) error {
```
Comment (ajm188) [inline go/cmd/internal/docgen/docgen.go:107]: weeeeeeeeeeeeee 🙃

File: go/cmd/internal/docgen/docgen.go [hunk 4]
```diff
@@ -79,9 +82,89 @@ func GenerateMarkdownTree(cmd *cobra.Command, dir string) error {
...
+			// Top-level command without children. Nothing to restructure.
+			continue
+		}
+	}
+
 	return nil
 }
 
+func newParentLinkSedCommand(parent string, file string) *exec.Cmd {
+	return exec.Command("sed", "-i.bak", "-e", fmt.Sprintf("s:(./%s/):(../):", parent), file)
```
Comment (mattlord) [inline go/cmd/internal/docgen/docgen.go:165]: We only delete the backup files, right? So we could also skip creating them:
```
return exec.Command("sed", "-i ''", "-e", fmt.Sprintf("s:(./%s/):(../):", parent), file)
```

**Review by mattlord — APPROVED** [review]
Thank you! ❤️

File: go/cmd/internal/docgen/docgen.go (hunk 4 above)
Comment (ajm188) [reply go/cmd/internal/docgen/docgen.go]: was trying out a bunch of different variants and got this one from a stack exchange post, but yours works!

**Pull request merged**